File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville:list, K)->int:
    heapq.heapify(scoville)
    count=0
    while True:
        if scoville[0]>=K:
            return count
        if len(scoville)==1:
            return -1
        one = heapq.heappop(scoville)
        two = heapq.heappop(scoville)
        new_s = one+two*2
        count +=1
        heapq.heappush(scoville,new_s)

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        PriorityQueue<Integer> q = new PriorityQueue<>();
        for(int s : scoville){
            q.add(s);
        }
        int count=0;
        while(q.size()>1){
            if (q.peek() >=K){
                return count;
            }
            Integer one =q.poll();
            Integer two =q.poll();
            count++;
            q.add(one+two*2);
        }
        
        if (q.peek()>=K){
            return count;
        }
        return -1;
    }
}